lib/rgeo/geos/utils.rb

```diff
@@ -49,9 +49,9 @@ def ffi_compute_dimension(geom)
           result
         end
 
-        def ffi_coord_seq_hash(cs, hash = 0)
-          (0...cs.length).inject(hash) do |h, i|
-            [hash, cs.get_x(i), cs.get_y(i), cs.get_z(i)].hash
+        def ffi_coord_seq_hash(coord_seq, hash = 0)
+          (0...coord_seq.length).inject(hash) do |_h, i|
+            [hash, coord_seq.get_x(i), coord_seq.get_y(i), coord_seq.get_z(i)].hash
```

Comment:
⚠️ 
Wow this one is a very bad mistake, we're lucky you edited it!

the `h` is actually necessary, and the `hash` usage is wrong here.

```suggestion
        def ffi_coord_seq_hash(coord_seq, init_hash = 0)
          (0...coord_seq.length).inject(init_hash) do |hash, i|
            [hash, coord_seq.get_x(i), coord_seq.get_y(i), coord_seq.get_z(i)].hash
```

Reproduction of the bug:

```bash
cat <<-RB | ruby -Ilib -rrgeo
f = RGeo::Geos.factory(native_interface: :ffi)
ls1 = f.line_string([f.point(1, 1), f.point(2, 2), f.point(3, 3)])
ls2 = f.line_string([f.point(1, 1), f.point(3, 3)])

exit 1 if ls1 == ls2
RB
```

I'm pushing this to the main branch ASAP as I think this is of major importance. This might also go to show that we don't have a lot of ffi users...

@keithdoggett FYI

Reply:
PR #344

Reply:
Let me know if I need to do anything here (like revert my changes) or wait until #344 is merged in.

Reply:
Merged now!